grc/blocks/json_config.block.yml

```diff
@@ -0,0 +1,37 @@
+id: json_config
+label: Json Config
+flags: [ show_id, python ]
+
+parameters:
+-   id: config_file
+    label: Config File
+    dtype: file_open
+    default: default
+-   id: schema_file
+    label: Config Schema
+    dtype: file_open
+    default: ""
+# Not my favorite thing because it doesn't explicitly close the file, but
+# it should be okay.  The garbage collector will take care of it.
+value: "${ json.load(open(config_file)) }"
+
+asserts:
+#- ${ schema_file == "" or validate(value, json.load(open(schema_file))) is None }
+- ${ schema_file == "" or validate(json.load(open(config_file)), json.load(open(schema_file))) is None }
+
+templates:
+    imports: |-
+        import json
+        from jsonschema import validate
+    var_make: "with open(${config_file}) as fid:\n\
+              \    self.${id} = ${id} = json.load(fid)\n\
+              self.${id}_schema = ${id}_schema = ${schema_file}\n\
+              if ${id}_schema:\n\
+              \    with open(${id}_schema) as fid:\n\
+              \        validate(${id}, json.load(fid))"
+
+documentation: |-
+    This block represents a json config file, that is read in as a dictionary.
```

Comment:
First time using the block, some questions pop in my head like 1) is the schema optional 2) is the json supposed to be only 1 level deep 3) you say it's read in as a dictionary but what does that mean in GR sense, like does each field become a variable? 

I think we should present people with the most basic example of using it, so for example:
1 - Make a new file called example.json with the contents:
{
  "amplitude": 0.1
}
2 - Add the JSON Config block and point Config File to that file
3 - Make a signal source, and for Amplitude enter the value json_config_0[ 'amplitude']
4 - Connect it to a Time Sink through a Throttle and you'll see the amplitude show up as 0.1

Reply:
Where should I put this example?  I could put it right in the doc string.  For example,
```
With an input file of 
{
  "samp_rate": 1e6,
  "center_freq": 8e9 
}
```
We can access the data, like `config["samp_rate"]`

Reply:
see https://wiki.gnuradio.org/index.php?title=JSON_Config#Example_Flowgraph

Reply:
Wouldn't hurt to also have an example json in the yaml as well, and it's fine if it's different example than the wiki

Reply:
Added in examples